akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -74,9 +76,10 @@ private[akka] final class InterceptorImpl[O, I](
     new InterceptorImpl(interceptor, newNested)
 
   override def receive(ctx: typed.TypedActorContext[O], msg: O): Behavior[O] = {
-    val interceptMessageType = interceptor.interceptMessageType
+    // TODO performance optimization could maybe to avoid isAssignableFrom if interceptMessageClass is Class[Object]?
```

Comment:
I think it would also be safe to have state in this class, so perhaps we could also cache when a type has been deemed assignable the first time? (If it shows up in benches I guess).

Reply:
the overhead of such cache lookup would probably be higher than isAssignableFrom (which is essentially an instanceof), but let's investigate if that really becomes a problem